lib/tasks/va_facilities.rake

```diff
@@ -145,11 +146,46 @@ namespace :va_facilities do
               puts "Updated facility: #{vaf['Official Station Name']}, #{ctr.to_s}"
             end
           end
+          current_facility_ids << facility.id if facility&.persisted?
+
         end
       end
+
+      non_current_facilities = VaFacility.where.not(id: current_facility_ids).pluck(:official_station_name, :id).to_h
+      current_date = Date.today.strftime("%Y-%m-%d")
+      filename = "#{Rails.root}/lib/assets/non_current_facility_ids_#{current_date}.json"
+
+      File.open(filename, "w") do |file|
+        file.write(non_current_facilities.to_json)
+      end
     puts "All VA facilities have been created or updated in the DB!"
+    puts "Non-current facility names and IDs saved to #{filename}."
   end
 
+  task :delete_non_current_va_facilities => :environment do
+    current_date = Date.today.strftime("%Y-%m-%d")
+    file_path = "#{Rails.root}/lib/assets/non_current_facility_ids_#{current_date}.json"
+
+    if File.exist?(file_path)
+      non_current_facilities = JSON.parse(File.read(file_path))
+      deletion_count = 0
+
+      non_current_facilities.each_value do |id|
+        facility = VaFacility.find_by(id: id)
+
+        if facility && facility.diffusion_histories.empty? && facility.practice_origin_facilities.empty?
+          name = facility.official_station_name
+          facility.delete
+          puts "Deleted facility: #{name}"
+          deletion_count += 1
+        end
```

Comment:
```suggestion
	else
	non_current_with_adoptions << facility.id
	...
        end
```
I think facilities requiring additional review would be collected somewhere on this end of the conditional